### ISSUE TITLE
Was wondering if you guys could point me in the right direction - really appreciate it.

### ISSUE THREAD

**Issue by lommaj** [magiclabs/react-native-demo#37]
Your demo app works fine but my pretty basic app throws an error in Android only that I cant seem to figure out.

its only when I have the relayer in the app.js

`<magic.Relayer/>`

My app is pretty basic - I am hoping I am just missing something simple.

```javascript
import React, { useState } from 'react'
import { SafeAreaProvider } from 'react-native-safe-area-context'
import { useUser } from 'stores/user'
import * as SecureStore from 'expo-secure-store'
import { Magic } from '@magic-sdk/react-native-expo'
import { Navigation } from 'navigation'

const magic = new Magic('omitted-for-issue-discussion')

export default function App() {
  const [loading, setLoading] = useState(true)

  const setToken = useUser(state => state.setDidToken)
  const getUser = useUser(state => state.getUser)

  React.useEffect(() => {
    async function loadResourcesAndDataAsync() {
      let token = await SecureStore.getItemAsync("alphaToken")

      if (token) {
        setToken(token)
        try {
          await getUser()
        } catch (e) {
          //token expired
        }
      }
      setLoading(false)
    }
    loadResourcesAndDataAsync()
  }, [])

  if (loading) {
    return null
  } else {
    return (
      <SafeAreaProvider>
        <magic.Relayer />
        <Navigation magic={magic} />
      </SafeAreaProvider>
    )
  }
}
```

```json
"dependencies": {
    "@magic-sdk/react-native-expo": "^14.3.0",
    "@react-navigation/drawer": "^6.5.8",
    "@react-navigation/native": "^6.1.3",
    "@react-navigation/native-stack": "^6.9.9",
    "expo": "~47.0.12",
    "expo-secure-store": "~12.0.0",
    "expo-status-bar": "~1.4.2",
    "nativewind": "^2.0.11",
    "react": "18.1.0",
    "react-native": "0.70.5",
    "react-native-gesture-handler": "^2.9.0",
    "react-native-reanimated": "~2.12.0",
    "react-native-safe-area-context": "^4.5.0",
    "react-native-screens": "~3.18.0",
    "react-native-webview": "^8.0.0",
    "unstated-next": "^1.1.0",
    "zustand": "^4.3.3"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9",
    "tailwindcss": "^3.2.4"
  },
```

The error I am getting is: 

`Invariant Violation: Failed to call into JavaScript module method .onMessage(). Module has not been registered as callable. Registered callable JavaScript modules (n = 12): Systrace, JSTimers, HeapCapture, SamplingProfiler, RCTLog, RCTDeviceEventEmitter, RCTNativeAppEventEmitter, GlobalPerformanceLogger, JSDevSupportModule, HMRClient, RCTEventEmitter, AppRegistry.`

**Comment by Ariflo** [comment]
Hey @lommaj thanks for point this out, we'll dig into it and see if there's anything we can do to prevent this on our side. 

In the meantime, we've found [this article](https://docs.expo.dev/troubleshooting/application-has-not-been-registered/) about the error from Expo to be immensely helpful in trying to track down the source of the `Invariant Violation` errors. 

Let us know if you learn anything new that'll help us find the fix.

**Comment by lommaj** [comment]
I just want to say thanks for pointing me to the docs. I have resolved the issue - seems like the out of the box react native starter has mis-matched dependencies and once I adjusted them to match the versions in your project and pegged them to that version - it worked fine.

I really appreciate your time and I apologize for not trying that first.

thanks for the great work!!  This example project really helped me get magic implemented!
